src/main/java/gyro/aws/ecs/EcsCapacityProviderResource.java

```diff
@@ -0,0 +1,265 @@
+            r -> r.name(getName())
+                .autoScalingGroupProvider(getAutoScalingGroupProvider().copyTo())
+        );
+
+        Wait.atMost(10, TimeUnit.MINUTES)
+            .checkEvery(30, TimeUnit.SECONDS)
+            .prompt(false)
+            .until(() -> isActive(client));
+    }
+
+    @Override
+    public void update(GyroUI ui, State state, Resource current, Set<String> changedFieldNames) throws Exception {
+        EcsClient client = createClient(EcsClient.class);
+
+        if (changedFieldNames.contains("tags")) {
+            EcsCapacityProviderResource currentResource = (EcsCapacityProviderResource) current;
+            Set<String> currentKeys = currentResource.getTags().keySet();
+
+            if (!currentKeys.isEmpty()) {
+                client.untagResource(r -> r.resourceArn(getArn()).tagKeys(currentKeys));
+            }
+
+            if (!getTags().isEmpty()) {
+                client.tagResource(
+                    r -> r.resourceArn(getArn()).tags(getTags().entrySet().stream()
+                        .map(o -> Tag.builder().key(o.getKey()).value(o.getValue()).build())
+                        .collect(Collectors.toList()))
+                );
+            }
+        }
+
+        Wait.atMost(10, TimeUnit.MINUTES)
+            .checkEvery(30, TimeUnit.SECONDS)
+            .prompt(false)
+            .until(() -> isActive(client));
+    }
+
+    @Override
+    public void delete(GyroUI ui, State state) throws Exception {
+
+    }
+
+    private CapacityProvider getCapacityProvider(EcsClient client) {
+        if (ObjectUtils.isBlank(getName())) {
+            throw new GyroException("'name' is missing, unable to load capacity provider.");
+        }
+
+        CapacityProvider provider = null;
+
+        try {
+            DescribeCapacityProvidersResponse response = client.describeCapacityProviders(
+                r -> r.capacityProviders(getName()).includeWithStrings("TAGS")
+            );
+
+            if (response.hasCapacityProviders()) {
+                provider = response.capacityProviders().get(0);
+            }
+        } catch (EcsException ex) {
+            // ignore
+        }
+
+        return provider;
+    }
+
+    private boolean isActive(EcsClient client) {
+        CapacityProvider provider = getCapacityProvider(client);
+
+        return provider != null && provider.statusAsString().equals("ACTIVE");
+    }
+
+    @Override
+    public List<ValidationError> validate(Set<String> configuredFields) {
+        List<ValidationError> errors = new ArrayList<>();
+
+        if (configuredFields.contains("name")) {
+            if (getName().toLowerCase().startsWith("aws") || getName().toLowerCase().startsWith("ecs") || getName().toLowerCase().startsWith("fargate")) {
+                errors.add(new ValidationError(
+                    this,
+                    "name",
+                    "The capacity provider 'name' cannot be prefixed with 'aws', 'ecs', or 'fargate'."
```

Comment:
No need to mention the resource as the error is in the same resource configuration

```suggestion
                    "'name' cannot be prefixed with 'aws', 'ecs', or 'fargate'."
```